gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-choose-existing-page/api-documentation-choose-existing-page.component.ts

```diff
@@ -0,0 +1,146 @@
+import { GioBannerModule, GioConfirmDialogComponent, GioConfirmDialogData } from '@gravitee/ui-particles-angular';
+
+import { ApiDocumentationChoosePageListComponent } from '../api-documentation-choose-page-list/api-documentation-choose-page-list.component';
+import { Api, Page } from '../../../../../entities/management-api-v2';
+import { ApiV2Service } from '../../../../../services-ngx/api-v2.service';
+import { ApiDocumentationV2Service } from '../../../../../services-ngx/api-documentation-v2.service';
+import { SnackBarService } from '../../../../../services-ngx/snack-bar.service';
+
+@Component({
+  selector: 'api-documentation-choose-existing-page',
+  standalone: true,
+  imports: [
+    ApiDocumentationChoosePageListComponent,
+    MatButton,
+    MatDialogActions,
+    MatDialogClose,
+    MatDialogContent,
+    MatIcon,
+    MatIconButton,
+    MatCard,
+    GioBannerModule,
+  ],
+  templateUrl: './api-documentation-choose-existing-page.component.html',
+  styleUrl: './api-documentation-choose-existing-page.component.scss',
+})
+export class ApiDocumentationChooseExistingPageComponent implements OnInit {
+  api: Api;
+  pages: Page[];
+  isLoading = false;
+  selectedPageId: string;
+  isReadOnly: boolean;
+  private unsubscribe$: Subject<void> = new Subject<void>();
+
+  @Output() pageSelected = new EventEmitter<string>();
+  @Output() cancelled = new EventEmitter<void>();
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly router: Router,
+    private readonly apiV2Service: ApiV2Service,
+    private readonly apiDocumentationV2Service: ApiDocumentationV2Service,
+    private readonly snackBarService: SnackBarService,
+    private readonly matDialog: MatDialog,
+  ) {}
+
+  ngOnInit() {
+    this.isLoading = true;
+
+    this.apiV2Service
+      .get(this.activatedRoute.snapshot.params.apiId)
+      .pipe(
+        tap((api) => {
+          this.api = api;
+          this.isReadOnly = this.api.originContext?.origin === 'KUBERNETES';
+        }),
+        switchMap(() => this.apiDocumentationV2Service.getApiPages(this.api.id)),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe((res) => {
+        this.pages = res.pages.filter((p) => p.homepage !== true);
+        this.isLoading = false;
+      });
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.unsubscribe();
+  }
+
+  saveWithConfirmation() {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: 'Confirm Homepage Selection',
+          content: 'This choice is final. Once a page becomes the homepage, it will no longer be accessible in your Documentation Pages.',
+          confirmButton: 'Confirm',
+        },
+      })
+      .afterClosed()
+      .pipe(filter((confirmed) => !!confirmed))
+      .subscribe({
+        next: (_) => {
+          this.save();
```

Comment:
The calls within `this.save()` should be in the main thread after `filter((confirmed) => !!confirmed)` instead of in a separate observable thread.

Reply:
Done